### ISSUE TITLE
Allow Control + Option + Space to open the select menu on Safari when using a screen reader

### ISSUE THREAD

**Issue by GCHQ-Developer-847** [mi6/ic-ui-kit#1451]
## Summary 
Add keyboard functionality to enable the opening of the ic-select with Control + Option + Space on Safari when using a screen reader.

## 💬 Description 
When carrying out screen reader testing on the multi-select, the user doing the testing tried to open the multi-select using Control + Option + Space. They were using VoiceOver and Edge, and they said that they often use this method to click on things; [it appears to be quite common / expected behaviour](https://dequeuniversity.com/screenreaders/voiceover-keyboard-shortcuts#:~:text=Activate%20a%20link%20or%20form%20control).

However, after trying this myself after the session, I found that it did in fact work on Edge, and it also works on Chrome and Firefox (not too sure why it didn't work on Edge during the testing session!). But it does not work on Safari.

I tried Control + Option + Space on Safari with the single select as well and also found that it did not open the menu either, so it is a problem with the ic-select as a whole.

This ticket is to explore why it doesn't work. It would be worth finding out first whether it is a VoiceOver bug, or a Safari bug, or perhaps a setting that needs turning on in Safari to enable this keyboard combination to work. It may require code to be added to listen for these keys being pressed, but it should work in theory because it worked automatically on the other browsers without extra behaviour being added, so this should be the last resort.

**Note:** It is only expected that Control + Option + Space should open the menu when VoiceOver is turned on (it should not work when it isn't).

## 💰 Use value 
This will enhance the experience for screen reader users. They can still open the select variants using just Space or Enter but it sounds like Control + Option + Space is commonly used by screen reader users so it would be good to get this working on Safari as well.

## 📝 Acceptance Criteria  
*Given* that I have VoiceOver enabled and I am using Safari,
*When* I tab to a select component and press Control + Option + Space,
*Then* the dropdown menu should open.
